driver/src/main/java/org/neo4j/driver/internal/net/SocketConnection.java

```diff
@@ -69,15 +72,34 @@ public SocketConnection( BoltServerAddress address, SecurityPlan securityPlan, L
             this.responseHandler = new SocketResponseHandler();
         }
 
-        this.socket = new SocketClient( address, securityPlan, logger );
-        socket.start();
+        this.socket.start();
+    }
+
+    // for mocked socket testing
+    SocketConnection( SocketClient socket, Logger logger )
+    {
+        this.socket = socket;
+        this.logger = logger;
+
+        if( logger.isDebugEnabled() )
```

Comment:
Maybe extract this `if-else` in a method like `createResponseHandler(Logger)`, otherwise it is duplicated in both constructors.